c/memsafety/cmp-freed-ptr.c

```diff
@@ -10,12 +10,14 @@ int main() {
     struct T* y = NULL;
 
     y = malloc(sizeof(*y));
+    long int adressY = (long int) y;
```

Comment:
I would use `intptr _t` from `stdint.h` instead of `long int` (cf. `7.20.1.4` of C18).
The task includes `stdlib.h` already, so additional clutter from `stdint.h` is not that much of a downside.
This will make the c file more portable.